solana/programs/swap-layer/src/utils/relayer_fees.rs

```diff
@@ -213,14 +217,50 @@ mod test {
     }
 
     #[test]
-    fn test_zero_compound() {
+    fn test_compound_beyond_max() {
+        let base = 1_000;
+        let percentage = 2_000_000; // 200%
+        let compounded = compound(percentage, base);
+
+        assert_eq!(compounded, None);
+    }
+
+    #[test]
+    fn test_compound_at_max_int() {
+        let base = u64::MAX;
+        let percentage = u32::MAX;
```

Comment:
```suggestion
        let percentage = 1_000_000; // crate::MAX_BPS
```

Sorry I realized that with the upper bound that this test is going to short-circuit and not check the overflow part.
This is now the highest value allowed that gets us into the `else { }` branch.